container: support kmd (#4984)
algorand/go-algorand: docker/README.md

| Setting | Value |
| ------- | ----- |
| EndpointAddress | 0.0.0.0:8080 |
| EnableDeveloperAPI | true |

> ### Environment Variables

Reviewer: The explicit variable checks in the table below should be updated from `If set,` to something like `When set to 1,`

| NUM_ROUNDS    | If set on a private network, override default of 30000 participation keys. |
| TOKEN         | If set, overrides the REST API token. |
| ADMIN_TOKEN   | If set, overrides the REST API admin token. |
| KMD_TOKEN | If set along with `START_KMD`, override the KMD REST API token. Otherwise, it's set to `ADMIN_TOKEN`. |

> | START_KMD | If set, start kmd service with no timeout. THIS SHOULD NOT BE USED IN PRODUCTION. |

Reviewer: Update descriptions to reflect explicit checks, i.e.

```suggestion
| START_KMD | When set to 1, start kmd service with no timeout. THIS SHOULD NOT BE USED IN PRODUCTION. |
```

```bash
docker run --rm -it \
    -p 4190:8080 \
    -p 4191:7833 \
    -e NETWORK=mainnet \
    -e FAST_CATCHUP=1 \
    -e TELEMETRY_NAME=name \
    -e TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa \
    -e START_KMD=1 \
    -v ${PWD}/data:/algod/data/ \
    --name mainnet-container \
    algorand/algod:latest
```

Explanation of parts:

* `-p 4190:8080` maps the internal algod REST API to local port 4190.
* `-p 4191:7833` maps the internal kmd REST API to local port 4191.
* `-e NETWORK=` can be set to any of the supported public networks.

> * `-e TELEMETRY_NAME=` enables telemetry reporting to Algorand for network health analysis. The value of this variable takes precedence over the `name` attribute set in `/etc/algorand/logging.config`.
> * `-e FAST_CATCHUP=1` causes fast catchup to start shortly after launching the network.
> * `-e START_KMD=1` signals to entrypoint to start the kmd REST API (THIS SHOULD NOT BE USED IN PRODUCTION).
> * `-e TELEMETRY_NAME=` enables telemetry reporting to Algorand for network health analysis.

Reviewer: TELEMETRY_NAME listed twice
```suggestion
* `-e TELEMETRY_NAME=` enables telemetry reporting to Algorand for network health analysis. The value of this variable takes precedence over the `name` attribute set in `/etc/algorand/logging.config`.
* `-e FAST_CATCHUP=1` causes fast catchup to start shortly after launching the network.
* `-e START_KMD=1` signals to entrypoint to start the kmd REST API (THIS SHOULD NOT BE USED IN PRODUCTION).
```

Maybe all the bullets here should list the values? (i.e. 'name' and 'mainnet')

```bash
docker run --rm -it \
    -p 4190:8080 \

>     -p 4191:7833 \

Reviewer 2: I think this should be `-o 4191:7833` instead of `-p 4191:7833`?

    -e NETWORK=mainnet \
    -e FAST_CATCHUP=1 \
    -e TELEMETRY_NAME=name \
    -e TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa \
    -e START_KMD=1 \
    -v ${PWD}/data:/algod/data/ \
    --name mainnet-container \
    algorand/algod:latest
```

Explanation of parts:

* `-p 4190:8080` maps the internal algod REST API to local port 4190.

> * `-o 4191:7833` maps the internal kmd REST API to local port 4191.

Reviewer 2: ```suggestion
* `-p 4191:7833` maps the internal kmd REST API to local port 4191.
```
